services/datamanager/builtin/builtin.go

```diff
@@ -730,6 +729,17 @@ func pollFilesystem(ctx context.Context, wg *sync.WaitGroup, captureDir string,
 		case <-ctx.Done():
 			return
 		case <-t.C:
+			logger.Debug("Polling")
+			shouldDelete, err := checkFileSystemStats(ctx, captureDir, logger)
+			if err != nil {
+				return
+			}
+			if shouldDelete {
+				start := time.Now()
+				deleteFiles(ctx, syncer, captureDir, logger)
```

Comment:
Should also check and warning log any returned error here, similar to the above. That warning can include a duration, but then would only log "finished deleting files" in the success case.